platforms/mobile/react native.md

```diff
@@ -0,0 +1,149 @@
+# Langley Foxall React Native
+
+
+
+*   [Introduction](#introduction)
+*   [Which should I use?](#whichshouldiuse)
+*   [Expo](#expo)
+    *   [Prerequisite knowledge](#Prerequisiteknowledge)
+    *   [Advantages](#Advantages)
+    *   [Disadvantages](#Disadvantages)
+    *   [Over The Air Updates](#OverTheAirUpdates)
+    *   [Cases where Expo cannot be used](#CaseswhereExpocannotbeused)
+    *   [Detaching](#detaching)
+*   [Pure React Native](#purereactnative)
+    *   [Prerequisite knowledge](#Prerequisiteknowledge)
+    *   [Advantages](#Advantages)
+    *   [Disadvantages](#Disadvantages)
+    *   [Linking Libraries](#linkinglibraries)
+    *   [Native Code](#nativecode)
+    *   [Deploying](#deploying)
+*   [Code Standards](#codestandards)
+*   [Recommended Packages](#recommendedpackages)
+*   [Blacklisted Packages](#blacklistedpackages)
+*   [Useful Links](#usefullinks)
+
+## Introduction
+React Native is a mobile framework developed and maintained by Facebook that allows the developer to develop apps for Android and iOS simultaneously. React Native generates an Android Studio and Xcode that will automatically bundle a JavaScript file that will control a majority of the logic and UI of the app on build.
+
+An open source toolchain called [Expo](https://expo.io) has also been developed around React Native. Expo removes the requirement for prerequisite knowledge of Android and iOS development. Having only a JavaScript file that a ReactJS app can be built using the exp command line tools or the Expo XDE application. The JavaScript file is bundled and delivered over the air to the developers device or simulator running the Expo app. *Note: You are required to download the Expo app and create an account if you wish to develop this way.*
+
+Both Pure React Native and Expo applications can be built mostly like a regular ReactJS Application and most Node Modules will be compatible. A notable exception to this is Routing and Navigation. [See recommended packages](#recommended packages).
+
+## Which should I use? ##
+
+Below is be listed the recommended prerequisite knowledge for each platform as well as the advantages and disadvantages. Use cases where Expo cannot be used will also be listed below.
+
+## Expo ##
+The Expo SDK only requires a Web Development skill set as all code written is in JavaScript. This allows basic applications to be built quickly, and cross platform.
+
+To see what Expo is capable of and for the quick start guide [visit the Expo documentation](https://docs.expo.io/versions/v29.0.0/sdk).
+
+#### Prerequisite knowledge
+
+   * ReactJS
+   * CSS
```

Comment:
I think `CSS` should be changed for
> NativeCSS, which is inspired by CSS

Reply:
Changed. My only planned changes at the moment are Expo Deployment procedures and fixing the table of contents links. Any suggestions?

Reply:
We could go into a crazy amount of details, even come up with React Native standards to help others jump on projects easily and instantly understand what's going on. Maybe even write a package to implement some standardized components

Reply:
I think a quick start guide for both Expo and Pure React Native would be a great idea for new projects. Although I'm not sure what standards we could come up with specifically for React Native since code standards would be taken from the language used in that part of the app. Perhaps we should create some UX and UI standards taking inspiration from [Apple's Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/) and [Google's design guidelines](https://developer.android.com/design/).

I'm against boilerplates as a concept in any programming language as they lead to bloated software. But I do think that a custom cli for initialising react native projects where you choose if you want to use Expo and you can choose which node modules you want to initialise the project with out of a list of curated packages would be a good idea.